src/Client/Connection.h

```diff
@@ -83,8 +83,8 @@ class Connection : private boost::noncopyable
     Connection(const String & host_, UInt16 port_,
         const String & default_database_,
         const String & user_, const String & password_,
-        const String & cluster_,
-        const String & cluster_secret_,
+        const String & cluster_ = "",
+        const String & cluster_secret_ = "",
```

Comment:
This was the intention, since default values may hide bugs (this ConnectionPoolWithFailover/ConnectionPool/Connection is pretty complex)

Maybe the code that uses this may be fixed?

Reply:
Yes, I hate default values.

But there is another codebase that is located in another repository that is depend on our code and it is very inconvenient to modify.

Reply:
> This was the intention, since default values may hide bugs (this ConnectionPoolWithFailover/ConnectionPool/Connection is pretty complex)
> 
> Maybe the code that uses this may be fixed?

Ok, I'll do that.